src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -164,7 +164,9 @@ const SignOrExecuteForm = ({
     setAdvancedParams(data)
   }
 
-  const submitDisabled = !isSubmittable || isEstimating || !tx
+  const walletIsWrongChain = currentChain?.chainId != wallet?.chainId
+  const walletIsNoOwner = !safe.owners.some((owner) => owner.value === wallet?.address)
```

Comment:
Non-owners can execute transactions, thought. So I would suggest to do smth like 
```
  const walletCanSubmit = willExecute || isSignableBy(transaction, wallet?.address || '')
```

Reply:
However, if it's a new tx (no `txId` passed), your check will work fine.
So to break down:

* New tx – must be an owner
* Existing tx to be signed – needs to be an owner who hasn't signed yet
* Existing tx to be executed – can be anyone

Reply:
Thanks for the feedback! I added the suggestions and fixed the testcases.

I am not familiar enough with the domain logic here- does your check already cover all 3 requirements or do i need to check if `txId` is passed additionally?

Reply:
I think your current version should work as it is 👍